contracts/psm/LisUSDPool.sol

```diff
@@ -0,0 +1,332 @@
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../hMath.sol";
+import "../interfaces/IStakeLisUSDListaDistributor.sol";
+import "../interfaces/VatLike.sol";
+
+contract LisUSDPool is AccessControlUpgradeable, ReentrancyGuardUpgradeable, PausableUpgradeable {
+    using SafeERC20 for IERC20;
+
+    struct Pool {
+        address asset;
+        address distributor;
+        bool active;
+    }
+
+    address public lisUSD; // lisUSD address
+    mapping(address => uint256) public shares; // user's share
+    // pool => user => balance
+    mapping(address => mapping(address => uint256)) public poolBalanceOf;
+    // pool => lista distributor
+    mapping(address => Pool) public pools;
+    uint256 public totalShares; // total
+    string public name; // pool name
+    string public symbol; // pool symbol
+    uint256 public rate; // share to lisUSD rate when last update
+    uint256 public lastUpdate; // last tps update time
+    uint256 public duty; // rate per second
+    uint256 public MAX_DUTY; // max rate per second
+    address public earnPool; // earn pool address
+    address public vat; // vat address
+
+    bytes32 public constant MANAGER = keccak256("MANAGER"); // manager role
+    bytes32 public constant PAUSE = keccak256("PAUSE"); // pause role
+    uint256 constant public RAY = 10**27;
+
+    event Withdraw(address account, uint256 amount);
+    event Deposit(address account, uint256 amount);
+    event SetDuty(uint256 duty);
+    event SetMaxDuty(uint256 maxDuty);
+    event RegisterPool(address pool, address asset, address distributor);
+    event RemovePool(address pool);
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    modifier onlyEarnPool() {
+        require(earnPool == msg.sender, "only earnPool can call this function");
+        _;
+    }
+
+    /**
+     * @dev initialize contract
+     * @param _lisUSD lisUSD address
+     * @param maxDuty max rate per second
+     */
+    function initialize(
+        address _lisUSD,
+        address _vat,
+        uint256 maxDuty
+    ) public initializer {
+        require(_lisUSD != address(0), "lisUSD cannot be zero address");
+        require(_vat != address(0), "vat cannot be zero address");
+
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        _setupRole(MANAGER, msg.sender);
+
+        lisUSD = _lisUSD;
+        vat = _vat;
+
+        name = "lisUSD single staking Pool";
+        symbol = "sLisUSD";
+        MAX_DUTY = maxDuty;
```

Comment:
Generally we use all caps for constant and immutables,

Hard code the value or rename to `maxDuty`?